### PR TITLE
lint: Revamp our linting rules, mostly around doc comments

### PR DESCRIPTION
**Pull request by beorn7**
Several things done here:

- Set `max-issues-per-linter` to 0 so that we actually see all linter warnings and not just 50 per linter. (As we also set `max-same-issues` to 0, I assume this was the intention from the beginning.)

- Stop using the golangci-lint default excludes (by setting `exclude-use-default: false`. Those are too generous and don't match our style conventions. (I have re-added some of the excludes explicitly in this commit. See below.)

- Re-add the `errcheck` exclusion we have used so far via the defaults.

- Exclude the signature requirement `govet` has for `Seek` methods because we use non-standard `Seek` methods a lot. (But we keep other requirements, while the default excludes completely disabled the check for common method segnatures.)

- Exclude warnings about missing doc comments on exported symbols. (We used to be pretty adamant about doc comments, but stopped that at some point in the past. By now, we have about 500 missing doc comments. We may consider reintroducing this check, but that's outside of the scope of this commit. The default excludes of golangci-lint essentially ignore doc comments completely.)

- By stop using the default excludes, we now get warnings back on malformed doc comments. That's the most impactful change in this commit. It does not enforce doc comments (again), but _if_ there is a doc comment, it has to have the recommended form. (Most of the changes in this commit are fixing this form.)

- Improve wording/spelling of some comments in .golangci.yml, and remove an outdated comment.

- Leave `package-comments` inactive, but add a TODO asking if we should change that.

- Add a new sub-linter `comment-spacings` (and fix corresponding comments), which avoids missing spaces after the leading `//`.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -25,7 +25,12 @@ linters:
     - loggercheck
 
 issues:
+  max-issues-per-linter: 0
   max-same-issues: 0
+  # The default exclusions are too aggresive. For one, they essentially disable
+  # any linting on doc comments. We disable default exclusions here and add
+  # exclusions fitting our codebase further down.
```
Comment (aknuds1) [inline .golangci.yml:32]: [Nit]
```suggestion
  # The default exclusions are too aggressive. For one, they essentially disable
  # any linting on doc comments. We disable default exclusions here and add
  # exclusions fitting our codebase further down.
```

**Review by aknuds1 — APPROVED** [review]
LGTM! (modulo nit)

**Review by aknuds1 — APPROVED** [review]

File: tsdb/wlog/wlog.go [hunk 2]
```diff
@@ -38,8 +38,8 @@ import (
 )
 
 const (
-	DefaultSegmentSize = 128 * 1024 * 1024 // 128 MB
-	pageSize           = 32 * 1024         // 32KB
+	DefaultSegmentSize = 128 * 1024 * 1024 // DefaultSegmentSize is 128 MB.
+	pageSize           = 32 * 1024         // pageSize is 32KB.
```
Comment (bboreham) [inline tsdb/wlog/wlog.go:42]: This is worse.  Can we turn off the check that made you do it?

Comment (beorn7) [reply tsdb/wlog/wlog.go:42]: I don't feel strongly whether this is worse or better.

However, if we turn off the check that made me do this, we won't check at all that doc comments start with the name they are commenting.

If it is really important to keep things here as they were, I would propose to add a nolint pragma.

**Pull request merged**